component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -330,6 +331,21 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             succeededAttempt = validateWithBackUpCodes(context, userToken, authenticatedUser);
         }
 
+        if (succeededAttempt && isLocalUser) {
+            String username = authenticatedUser.toFullQualifiedUsername();
+            try {
+                String userEmail = getEmailValueForUsername(username, context);
+                if (StringUtils.isBlank(userEmail)) {
+                    Object verifiedEmailObject = context.getProperty(EmailOTPAuthenticatorConstants.REQUEST_USER_EMAIL);
+                    if (verifiedEmailObject != null) {
+                        updateEmailAddressForUsername(context, username);
+                    }
+                }
+            } catch (EmailOTPException e) {
+                throw new AuthenticationFailedException("Failed to get the email claim", e);
```

Comment:
lets add some context info.
Failed to get the email claim for user: + username

Reply:
Fixed the issue